libraries/core_libs/network/src/tarcap/shared_states/peers_state.cpp

```diff
@@ -134,4 +134,31 @@ bool PeersState::is_peer_malicious(const dev::p2p::NodeID& peer_id) {
   return false;
 }
 
+void PeersState::set_peer_light_node(const dev::p2p::NodeID& peer_id) {
+  light_node_peers_.emplace(peer_id, std::chrono::steady_clock::now());
+}
+
+bool PeersState::is_peer_light_node(const dev::p2p::NodeID& peer_id) {
+  // Peers are marked light nodes for the time defined in conf_.network_peer_light_node_timeout
```

Comment:
I don't understand why define light node peers in timing. 
In this way, some light nodes will be treat as light nodes, and some will be not. 
What is the purpose for that?

Reply:
it is possible for a node to change its state and go from light node back to full node

Reply:
to go from light node to full node, user have to change config and restart the node, in which case the node will create new connections as full node, so `network_peer_light_node_timeout` seems like we could delete it ?

Reply:
exactly, this `light_node` flag should be inside Peer class as once the new node connects it is either light or full node. It cannot change it's state without restarting the node. So let's put it into the Peer class and then you dont need no timeout either

Reply:
Implemented